sangria/sangria_2.13/src/main/scala/com/linecorp/armeria/server/sangria/SangriaGraphqlService.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.sangria
+
+import com.fasterxml.jackson.databind.{JsonNode, ObjectMapper}
+import com.linecorp.armeria.common.HttpStatus.{BAD_REQUEST, INTERNAL_SERVER_ERROR}
+import com.linecorp.armeria.common.annotation.UnstableApi
+import com.linecorp.armeria.common.{HttpHeaders, HttpResponse}
+import com.linecorp.armeria.internal.server.JacksonUtil
+import com.linecorp.armeria.scala.implicits._
+import com.linecorp.armeria.server.graphql.protocol.{AbstractGraphqlService, GraphqlRequest}
+import com.linecorp.armeria.server.sangria.SangriaGraphqlService.mapper
+import com.linecorp.armeria.server.sangria.SangriaJackson._
+import com.linecorp.armeria.server.{HttpService, ServiceRequestContext}
+import sangria.execution._
+import sangria.execution.deferred.DeferredResolver
+import sangria.parser.{QueryParser, SyntaxError}
+import sangria.schema.Schema
+import sangria.slowlog.SlowLog
+import sangria.validation.QueryValidator
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+private final class SangriaGraphqlService[Ctx, Val](
```

Comment:
Should we make this class public and make `build()` and `apply()` return `SangriaGraphqlService`?

Reply:
That looks better.

Reply:
I forget why I make `build()` and `apply()` return `HttpService`. 
If we return `SangriaGraphqlService[Ctx, Val]`, Scala compiler fails to compile the example project with the following error:
```
[Error] /Users/ikhun/src/armeria/examples/graphql-sangria/src/main/scala/example/armeria/server/sangria/Main.scala:34: overloaded method service with alternatives:
  (serviceWithRoutes: com.linecorp.armeria.server.HttpServiceWithRoutes,decorators: java.util.function.Function[_ >: com.linecorp.armeria.server.HttpService, _ <: com.linecorp.armeria.server.HttpService]*)com.linecorp.armeria.server.ServerBuilder <and>
  (serviceWithRoutes: com.linecorp.armeria.server.HttpServiceWithRoutes,decorators: Iterable[_ <: java.util.function.Function[_ >: com.linecorp.armeria.server.HttpService, _ <: com.linecorp.armeria.server.HttpService]])com.linecorp.armeria.server.ServerBuilder <and>
  (route: com.linecorp.armeria.server.Route,service: com.linecorp.armeria.server.HttpService)com.linecorp.armeria.server.ServerBuilder <and>
  (pathPattern: String,service: com.linecorp.armeria.server.HttpService)com.linecorp.armeria.server.ServerBuilder
 cannot be applied to (String, com.linecorp.armeria.server.sangria.SangriaGraphqlService[example.armeria.server.sangria.UserRepo,Unit])
two errors found
```

I can avoid the weird compile error by adding a non-varargs version of the method.

Reply:
`@FunctionInterface` with varargs is a hidden pitfall for Scala. 😭

Reply:
What's the status of this compilation error? It looks like this PR doesn't have a compilation error anymore.

Reply:
I forgot to update how to fix it.
I didn't find out why Scala compiler produced the error message. 
But I realized that I missed adding `implementation project(':graphql-protocol')` to the example project.
After I add the `:graphql-protocol` dependency, the compiler error magically was gone. 😅